api/v1alpha1/virtualmachine_types.go

```diff
@@ -484,6 +511,17 @@ func (vm VirtualMachine) NamespacedName() string {
 	return vm.Namespace + "/" + vm.Name
 }
 
+// VirtualMachineTemplate defines the specification for configuring VirtualMachine Template.
+// VirtualMachineTemplate is used during VM customization to populate VM OVF properties.
+// VirtualMachineTemplate is utilized for using Golang-based templating to provide access to dynamic configuration data.
+type VirtualMachineTemplate struct {
+	// NetworkSpec defines the specification of Network
+	Net NetworkSpec `json:"network"`
```

Comment:
Maybe use pointer for Net and VM? And the comment should match the variable name, not the type name.